MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -2185,6 +2185,64 @@ SetAccessAttributesInMemoryMap (
   return EFI_SUCCESS;
 }
 
+/**
+  Merge contiguous memory map entries with the same attributes.
+
+  @param  MemoryMap              A pointer to the memory map
+  @param  MemoryMapSize          A pointer to the size, in bytes, of the
+                                 MemoryMap buffer. On input, this is the size of the current
+                                 memory map.  On output, it is the size of new memory map after merge.
+  @param  DescriptorSize         Size, in bytes, of an individual EFI_MEMORY_DESCRIPTOR.
+**/
+STATIC
+VOID
+MergeMemoryMapByAttribute (
+  IN OUT EFI_MEMORY_DESCRIPTOR  *MemoryMap,
+  IN OUT UINTN                  *MemoryMapSize,
+  IN CONST UINTN                *DescriptorSize

```

Comment:
DescriptorSize ...
EFI_MEMORY_DESCRIPTOR shouldn't really need to change.
The other functions in this file all use "UINTN DescriptorSize", why does this one need to be a pointer?

EFI_MEMORY_DESCRIPTOR  comes from UefiSpec.h, do we even need the size of EFI_MEMORY_DESCRIPTOR  paassed or could you just embed sizeof(EFI_MEMORY_DESCRIPTOR ) into the code?

Reply:
The actual size purposefully isn't set to sizeof(EFI_MEMORY_DESCRIPTOR): https://github.com/microsoft/mu_basecore/blob/1fba347ad4bc2f0a399c388054c8fd597e302fe9/MdeModulePkg/Core/Dxe/Mem/Page.c#L1822